File: PayFabric/Samples/net40/Wallet/Update.cs
// =====================================================================
//  THIS CODE AND INFORMATION ARE PROVIDED "AS IS" WITHOUT WARRANTY OF ANY
//  KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
//  IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
//  PARTICULAR PURPOSE.
// =====================================================================

using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.IO;
using System.Net;
using System.Runtime.Serialization;

namespace Samples.Net40
{
    /// <summary>
    /// This sample is to demo how to update an existing card
    /// </summary>
    public partial class Wallet
    {
        /// <summary>
        /// Card number is not able to be updated. But you can remove the old card and add the new one.
        /// Below example is updating expiration date, billto city, and 2 user defined fields
        /// </summary>
        /// <param name="cardId">Card guid</param>
        public void Update(Guid cardId)
        {
            try
            {
                //  Populate POST String
                StringBuilder datastring = new StringBuilder();
                datastring.Append("{");
                datastring.Append("\"ID\":\"d8f265e3-fbae-459d-b04b-857679d35c84\",");
                datastring.Append("\"ExpDate\":\"0219\",");
                datastring.Append("\"CardHolder\":{");
                datastring.Append("\"FirstName\":\"newupdate\",");
                datastring.Append("\"MiddleName\":\"\",");
                datastring.Append("\"LastName\":\"newupdate\"");
                datastring.Append("},");
                datastring.Append("\"Billto\":{");
                datastring.Append("\"Zip\":\"123456\",");
                datastring.Append("\"Country\":\"new\",");
                datastring.Append("\"State\":\"new\",");
                datastring.Append("\"Email\":\"\",");
                datastring.Append("\"City\":\"Rowland Height\",");
                datastring.Append("\"Line1\":\"Fullerton Blvd\"");
                datastring.Append("},");
                datastring.Append("\"UserDefined1\":\"New Update\",");
                datastring.Append("\"UserDefined2\":\"New Update\"");
                datastring.Append("}");

                // POST
                byte[] data = System.Text.Encoding.UTF8.GetBytes(datastring.ToString());
                var url = "https://sandbox.payfabric.com/payment/api/wallet/update";
                ServicePointManager.SecurityProtocol = SecurityProtocolType.Tls12;
                HttpWebRequest httpWebRequest = WebRequest.Create(url) as HttpWebRequest;
                httpWebRequest = WebRequest.Create(url) as HttpWebRequest;
                httpWebRequest.ContentType = "application/json; charset=utf-8";
                httpWebRequest.Headers["authorization"] = new Token().Create();
                httpWebRequest.Method = "POST";
                httpWebRequest.ContentLength = data.Length;
                Stream stream = httpWebRequest.GetRequestStream();
                stream.Write(data, 0, data.Length);
                stream.Close();
                HttpWebResponse httpWebResponse = httpWebRequest.GetResponse() as HttpWebResponse;
                Stream responseStream = httpWebResponse.GetResponseStream();
                StreamReader streamReader = new StreamReader(responseStream);
                string result = streamReader.ReadToEnd();
                streamReader.Close();
                responseStream.Close();
                httpWebRequest.Abort();
                httpWebResponse.Close();

                //
                // Sample response
                // ----------------------------------------------------
                //{
                //    "Result":"true"  
                //}
                // ----------------------------------------------------
                //

            }
            catch (WebException e)
            {
                //  Handling exception from PayFabric
            }
            catch (Exception e)
            {
                //  Handling exception
            }
        }
    }
}

File: PayFabric/Samples/java/Wallet/Update.java
package com.nodus.payfabric.samples.wallet;

import java.io.BufferedReader;
import java.io.DataOutputStream;
import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;
import java.net.URL;
import java.util.UUID;

import javax.net.ssl.HttpsURLConnection;

import com.nodus.payfabric.samples.misc.Token;

/**
 * This sample is to demo how to update an existing card
 * */
public class Update {

	/**
	 * Card number is not able to be updated. But you can remove the old card
	 * and add the new one. Below example is updating expiration date, billto
	 * city, and 2 user defined fields
	 * 
	 * @param cardID
	 *            Card UUID
	 * */
	public void updateCard(UUID cardID) {

		// Populate POST String
		StringBuilder datastring = new StringBuilder();
		datastring.append("{");
		datastring.append("\"ID\":\"" + cardID.toString() + "\",");
		datastring.append("\"ExpDate\":\"0219\",");
		datastring.append("\"Billto\":{");
		datastring.append("\"City\":\"Rowland Height\",");
		datastring.append("\"Line1\":\"Fullerton Blvd\"");
		datastring.append("},");
		datastring.append("\"UserDefined1\":\"New Update\",");
		datastring.append("\"UserDefined2\":\"New Update\"");
		datastring.append("}");
		
		System.out.println(datastring);

		// POST
		try {

			byte[] data = datastring.toString().getBytes("UTF-8");
			String url = "https://sandbox.payfabric.com/payment/api/wallet/update";
			URL obj = new URL(url);
			HttpsURLConnection con = (HttpsURLConnection) obj.openConnection();
			con.setRequestMethod("POST");
			con.setRequestProperty("Content-Type",
					"application/json; charset=utf-8");
						
			// Replace with your own device id and device password
			con.setRequestProperty("authorization", "0ad64468-f4bc-0c99-4e31-bd08dd862c43|123456abc");
			
			con.setRequestProperty("Content-Length", data.length + "");
			con.setDoOutput(true);
			DataOutputStream streamWriter = new DataOutputStream(
					con.getOutputStream());
			streamWriter.write(data);
			streamWriter.flush();
			streamWriter.close();
			InputStream stream;
			int responseCode = con.getResponseCode();
			if (responseCode >= 400) {
				stream = con.getErrorStream();
			} else {
				stream = con.getInputStream();
			}
			BufferedReader streamReader = new BufferedReader(
					new InputStreamReader(stream));
			String inputLine;
			StringBuffer response = new StringBuffer();

			while ((inputLine = streamReader.readLine()) != null) {
				response.append(inputLine);
			}
			streamReader.close();
			con.disconnect();
			System.out.println(response.toString());

			if (responseCode >= 400) {

				// Handling exception from PayFabric
			}

			//
			// Sample response
			// ----------------------------------------------------
			// {
			// "Result":"true"
			// }
			// ----------------------------------------------------
			//

		} catch (IOException e) {
			System.out.println(e.getMessage());

			// Handling exception
		}
	}
}
